hal/src/stm32f2xx/static_recursive_mutex.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "hal_irq_flag.h"
+#include "debug.h"
+
+#include "FreeRTOSConfig.h"
+#include "FreeRTOS.h"
+#include "semphr.h"
+
+#if PLATFORM_ID == 6 || PLATFORM_ID == 8
+#include "wwd_rtos_interface.h"
+#endif // PLATFORM_ID == 6 || PLATFORM_ID == 8
+
+class StaticRecursiveMutex {
+public:
+    StaticRecursiveMutex() :
+            mutex_(nullptr) {
+        init();
+    }
+
+    ~StaticRecursiveMutex() {
+        if (mutex_) {
+            vSemaphoreDelete(mutex_);
+        }
+    }
+
+    bool lock() {
+        if (!mutex_) {
+            init();
+        }
+        return (xSemaphoreTakeRecursive(mutex_, portMAX_DELAY) == pdTRUE);
```

Comment:
I still think we should add a timeout here at least in debug builds to make it easier to spot incorrect usage of DCT. That way the timeout will trigger an assert in `dct_lock` instead of a deadlock.